Objects/longobject.c

```diff
@@ -3676,7 +3676,20 @@ long_hash(PyObject *obj)
     }
     i = _PyLong_DigitCount(v);
     sign = _PyLong_NonCompactSign(v);
-    x = 0;
+
+    // unroll first two digits
+#if ( PyHASH_BITS > PyLong_SHIFT )
+    assert(i>=2);
+    --i;
+    x = v->long_value.ob_digit[i];
+    assert(x < PyHASH_MODULUS);
+#endif
+#if ( PyHASH_BITS > (2 * PyLong_SHIFT) )
+    --i;
+    x = ((x << PyLong_SHIFT));
```

Comment:
Drop redundant parenthesis or just
```suggestion
    x <<= PyLong_SHIFT;
```
?